future-apps/agent/contracts/standards/ERC1271.sol

```diff
@@ -0,0 +1,41 @@
+pragma solidity 0.4.24;
+
+
+// ERC1271 on Feb 12th, 2019: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1271.md
+// Using `isValidSignature(bytes32,bytes)` even though the standard still hasn't been modified
+// Rationale: https://github.com/ethereum/EIPs/issues/1271#issuecomment-462719728
+contract ERC1271 {
+    // bytes4(keccak256("isValidSignature(bytes,bytes)")
+    bytes4 constant internal ERC1271_RETURN_VALID_SIGNATURE =   0x20c13b0b;
+    bytes4 constant internal ERC1271_RETURN_INVALID_SIGNATURE = 0x00000000;
+
+    /**
+    * @dev Function must be implemented by deriving contract
+    * @param _hash Arbitrary length data signed on the behalf of address(this)
+    * @param _signature Signature byte array associated with _data
+    * @return A bytes4 magic value 0x20c13b0b if the signature check passes, 0x00000000 if not
+    *
+    * MUST NOT modify state (using STATICCALL for solc < 0.5, view modifier for solc > 0.5)
+    * MUST allow external calls
+    */ 
+    function isValidSignature(bytes32 _hash, bytes memory _signature) public view returns (bytes4);
+
+    function isValidSignatureReturn(bool isValid) internal pure returns (bytes4) {
```

Comment:
I'd name this like `returnIsValidSignatureBytes()` or something. Otherwise it looks too similar to `isValidSignature()`.